tests/clv/models/test_pareto_nbd.py

```diff
@@ -394,15 +372,262 @@ def test_save_load_pareto_nbd(self):
+        )
+        assert (res_zero > res_high_purchase2).all()
+
+        # Probability should go down if dropout covariate goes up (coefficient is positive)
+        test_data_alt = test_data_zero.copy()
+        test_data_alt["dropout_cov"] += 1
+        res_high_drop = model.expected_purchase_probability(test_data_alt).mean(
+            ("chain", "draw")
+        )
+        assert (res_zero > res_high_drop).all()
+
+    def test_distribution_method(self):
+        model = self.model_with_covariates
+
+        reps = 100
+        test_data_zero = pd.DataFrame(
+            {
+                "customer_id": range(3 * reps),
+                "frequency": [1, 2, 0] * reps,
+                "recency": [7, 5, 2] * reps,
+                "purchase_cov1": [0, 0, 0] * reps,
+                "purchase_cov2": [0, 0, 0] * reps,
+                "dropout_cov": [0, 0, 0] * reps,
+                "T": [20, 20, 20] * reps,
+                "future_t": [2, 3, 4] * reps,
+                "n_purchases": [2, 1, 4] * reps,
+            }
+        )
+
+        # Probability should match model without covariates, when covariates are all zero
+        res_zero = model.predictive_distribution(test_data_zero).mean(("chain", "draw"))
+        # res_zero_ref = self.model_without_covariates.predictive_distribution(test_data_zero).mean(
+        #     ("chain", "draw"))
+        # np.testing.assert_allclose(res_zero["dropout"].mean("customer_id"), res_zero_ref["dropout"], rtol=0.3)
+        # np.testing.assert_allclose(res_zero["purchase_rate"].mean("customer_id"), res_zero_ref["purchase_rate"], rtol=0.3)
+        # np.testing.assert_allclose(
+        #     res_zero["recency_frequency"].sel(obs_var="recency").mean("customer_id"),
+        #     res_zero_ref["recency_frequency"].sel(obs_var="recency").mean("customer_id"),
+        #     rtol=0.3
+        # )
+        # np.testing.assert_allclose(res_zero["recency_frequency"].sel(obs_var="frequency").mean("customer_id"),
+        #                            res_zero_ref["recency_frequency"].sel(obs_var="frequency").mean("customer_id"),
+        #                            rtol=0.3)
+
+        # Test case where transaction behavior should increase
+        test_data_alt = test_data_zero.copy()
+
+        test_data_alt[
+            "purchase_cov1"
+        ] += 1  # more likely to purchase (positive coefficient)
+        test_data_alt[
+            "purchase_cov2"
+        ] -= 1  # more likely to purchase (negative coefficient)
+        test_data_alt[
+            "dropout_cov"
+        ] -= 1  # less likely to drop out (positive coefficient)
+        res_high_transaction = model.predictive_distribution(test_data_alt).mean(
+            ("chain", "draw")
+        )
+        # FIXME: Test is failing for the first two conditions, signs seem swapped?
+        assert (res_zero["purchase_rate"] < res_high_transaction["purchase_rate"]).all()
+        assert (res_zero["dropout"] > res_high_transaction["dropout"]).all()
```

Comment:
@ColtAllen These two checks fail while the two below pass. Are we perhaps taking draws of the reciprocal of these quantities?

Reply:
Yes; these two lines should not be reciprocals:

https://github.com/ricardoV94/pymc-marketing/blob/6ba6ee1e7db2a93467beea974079fb0039cafca8/pymc_marketing/clv/models/pareto_nbd.py#L795

https://github.com/ricardoV94/pymc-marketing/blob/6ba6ee1e7db2a93467beea974079fb0039cafca8/pymc_marketing/clv/models/pareto_nbd.py#L801

Reply:
Fixed